internal/provider/contract_managment.go

```diff
@@ -0,0 +1,94 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/threefoldtech/substrate-client"
+)
+
+func IsValidNameContract(sub *substrate.Substrate, contractID uint64) (bool, error) {
```

Comment:
does valid here means existing contract? or non existing contract? because created contracts to me are `valid`, if you mean something else, please update the function name